### PR TITLE
Auto update widths of inputs on type

### PR DESCRIPTION
**Pull request by SEAjamieD**
### :nut_and_bolt: Description: What code changed, and why?
By default, the number inputs on data-lifecycle were only large enough to show two digits.  We will now auto update the width of the input upon typing so that the entire number is always visible.

### :chains: Related Resources
Fixes: https://github.com/chef/automate/issues/4040

### :+1: Definition of Done
Data Lifecycle input values are always visible to the user, regardless of how long.

### :athletic_shoe: How to Build and Test the Change
build components/automate-ui-devproxy && start_all_services
make serve

navigate to https://a2-dev.test/settings/data-lifecycle
In any/all of the type-able number inputs, type as many valid numbers as you wish and view the input expand/shrink as necessary.

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
auto resize

### CODE REVIEW

**Comment by SEAjamieD** [comment]
added a small update to run after the for is initially populated.  Noticed that when a large number is saved in the database, the input width was too small.

File: components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts [hunk 1]
```diff
@@ -178,6 +179,19 @@ export class AutomateSettingsComponent implements OnInit, OnDestroy {
     return allowedKeys.includes(key);
   }
 
+  // Update the width of input when greater than one number
```
Comment (msorens) [inline components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts:182]: Much less confusing this way:
```suggestion
  // Update the width of input when greater than one digit
```

File: components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts [hunk 2]
```diff
@@ -178,6 +179,19 @@ export class AutomateSettingsComponent implements OnInit, OnDestroy {
...
+  public autoUpdateInputWidth(element: HTMLInputElement): void {
+    // Keep default in sync with input width in automate-settings.component.scss
+    const DEFAULT_WIDTH = 64;
```
Comment (msorens) [inline components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts:185]:
```suggestion
    const DEFAULT_WIDTH_PIXELS = 64;
```

File: components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts [hunk 3]
```diff
@@ -178,6 +179,19 @@ export class AutomateSettingsComponent implements OnInit, OnDestroy {
...
+    const length = element.value.length;
+
+    if (length > 1) {
+      element.style.width = `${DEFAULT_WIDTH + (9 * (length - 1))}px`;
```
Comment (msorens) [inline components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts:189]: We are guaranteed that characters are 9 pixels wide or is this an approximation?

File: components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts [hunk 4]
```diff
@@ -178,6 +179,19 @@ export class AutomateSettingsComponent implements OnInit, OnDestroy {
     return allowedKeys.includes(key);
   }
 
+  // Update the width of input when greater than one number
+  public autoUpdateInputWidth(element: HTMLInputElement): void {
+    // Keep default in sync with input width in automate-settings.component.scss
+    const DEFAULT_WIDTH = 64;
+    const length = element.value.length;
```
Comment (msorens) [inline components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts:186]:
```suggestion
    const numDigits = element.value.length;
```

File: components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts (hunk 3 above)
Comment (SEAjamieD) [reply components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts:189]: this is an approximation that errs on the side of wider

**Comment by SEAjamieD** [comment]
Not to self - onChange wasn't totally covering what was necessary.  Opened to card to update when possible: https://github.com/chef/automate/issues/4175

Making a directive as Scott mentioned above could be great too.

**Pull request merged**